src/main/kotlin/com/jetbrains/snakecharm/codeInsight/SnakemakeAPI.kt

```diff
@@ -72,85 +75,89 @@ object SnakemakeAPI {
      * Also see [ImplicitPySymbolsProvider], it also processes 'InputFiles', etc. symbols
      */
     val SECTION_ACCESSOR_CLASSES = mapOf(
-        "snakemake.io.InputFiles" to "input",
-        "snakemake.io.OutputFiles" to "output",
-        "snakemake.io.Params" to "params",
-        "snakemake.io.Log" to "log",
-        "snakemake.io.Resources" to "resources"
```

Comment:
Looks like it's a snakemake bug, because i got the semae error in simpler example:
```
use rule name as new_name with:
    output: "file.txt"
    threads: 4
```
But that works fine:
```
use rule name as new_name with:
    output: 
        "file.txt"
    threads: 4
```
The same with your example:
```
use rule rule from other_355_workflow with: # Works fine too
    input: 
        "rule_355.input"
    benchmark: "echo.txt"
```

Reply:
Ok, agree, is snakemake bug. 

P.S: In snakemake is parsed using: `...../python3.8/site-packages/snakemake/parser.py`
Block `use` is parsed by `UseRule`, unsupported keywords error is thrown from method `UseRule.block_content(..)`.
```python
    def block_content(self, token):
        if is_comment(token):
            yield "\n", token
            yield token.string, token
        elif is_name(token):
            try:
                self._with_block.extend(self.subautomaton(token.string).consume())
                yield from ()
            except KeyError:
                self.error(
                    "Unexpected keyword {} in rule definition".format(token.string),
                    token,
                )
            except StopAutomaton as e:
                self.indentation(e.token)
                self.block(e.token)
        else:
            self.error(
                "Expecting a keyword or comment "
                "inside a 'use rule ... with:' statement.",
                token,
            )

```

 AFAIU supported sections is taken from `UseRule:  subautomata = rule_property_subautomata`,i.e:
```
rule_property_subautomata = dict(
    name=Name,
    input=Input,
    output=Output,
    params=Params,
    threads=Threads,
    resources=Resources,
    priority=Priority,
    version=Version,
    log=Log,
    message=Message,
    benchmark=Benchmark,
    conda=Conda,
    singularity=Singularity,
    container=Container,
    containerized=Containerized,
    envmodules=EnvModules,
    wildcard_constraints=WildcardConstraints,
    shadow=Shadow,
    group=Group,
    cache=Cache,
    handover=Handover,
)
```

Reply:
Thus looks like everything except execution sections. As a result for `shell` we get error like `Unexpected keyword shell in rule definition (rule_355_use.smk, line 21)` , so better error message will be provided by issue https://github.com/JetBrains-Research/snakecharm/issues/408